src/app/containers/CpsRecommendations/index.jsx

```diff
@@ -1,57 +1,159 @@
 import React, { useContext } from 'react';
-import { shape, arrayOf, number, string } from 'prop-types';
-import { Headline, Link } from '@bbc/psammead-story-promo';
+import { arrayOf, shape, number } from 'prop-types';
+import styled from 'styled-components';
+import path from 'ramda/src/path';
+import StoryPromo, { Headline, Link } from '@bbc/psammead-story-promo';
+import { StoryPromoLiBase, StoryPromoUl } from '@bbc/psammead-story-promo-list';
+import { C_LUNAR, C_GHOST } from '@bbc/psammead-styles/colours';
+import Image from '@bbc/psammead-image';
+import {
+  GEL_SPACING,
+  GEL_SPACING_HLF,
+  GEL_SPACING_DBL,
+} from '@bbc/gel-foundations/spacings';
+import { GEL_GROUP_3_SCREEN_WIDTH_MAX } from '@bbc/gel-foundations/breakpoints';
 
+import { storyItem } from '#models/propTypes/storyItem';
 import { ServiceContext } from '#contexts/ServiceContext';
 import useToggle from '#hooks/useToggle';
+import CpsOnwardJourney from '../CpsOnwardJourney';
 import Grid from '../../components/Grid';
 
+const StoryPromoWrapper = styled(StoryPromo)`
+  display: grid;
+  margin: ${GEL_SPACING_HLF} ${GEL_SPACING_DBL};
+  background-color: ${C_GHOST};
+  @media (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
+    margin: ${GEL_SPACING_HLF} 0;
+  }
+`;
+
+const RecommendationsWrapper = styled.div`
+  background-color: ${C_LUNAR};
+  padding-bottom: ${GEL_SPACING};
+`;
+
+const getPromoItemProps = item => ({
+  imgSrc: path(['indexImage', 'path'], item),
+  href: path(['locators', 'assetUri'], item),
+  headline: path(['headlines', 'headline'], item),
+  alt: path(['indexImage', 'altText'], item),
+});
+
 const CpsRecommendations = ({ items, parentColumns }) => {
-  const { recommendations, script, service } = useContext(ServiceContext);
+  const { recommendations, script, service, dir } = useContext(ServiceContext);
   const { enabled } = useToggle('cpsRecommendations');
 
   const { hasStoryRecommendations } = recommendations;
 
-  if (!hasStoryRecommendations || !enabled) return null;
+  if (!hasStoryRecommendations || !enabled || !items.length) return null;
 
-  return (
+  const singleTransform = item => {
```

Comment:
- What does `singleTransform` mean? What's being "transformed" here?
- Is there a reason this is not not a React component in it's own module?

Reply:
@simonsinclair the `singleTransform` and `listTransform` are just conventions that was used originally in `CpsRelatedContent` which was set in the by the media team as they originally created the onward journey container. I think that same convention bled out into the top stories, F&A sections. I just followed that same convention as the `CpsOnwardJourney` container which we're reusing is tightly dependent on.

Reply:
If we're not going to fix this here, would you mind creating an issue to separate these functions into React components? 

The issue should make clear that although these functions happen to be named `singleTransform` and `listTransform`, they return JSX, and should be treated (and named) more like React components. They're missing prop-validation and unit tests. They force the `CpsRecommendations` to import components like `Header`, `Link` and `Image`, which it has no concerns with. These components should be passed directly to `CpsOnwardJourney`'s "Transform" props.

Reply:
@simonsinclair created https://github.com/bbc/simorgh/issues/7289 to cater for this. Please have a look and let me know what you think.

Reply:
Thanks, Rowland. It looks great.